core/src/main/java/hudson/model/Computer.java

```diff
@@ -1103,7 +1105,7 @@ protected void removeExecutor(final Executor e) {
                         ciBase.removeComputer(Computer.this);
                     }
                 } else if (isIdle()) {
-                    Listeners.notify(ComputerListener.class, false, l -> l.onIdle(this));
+                    threadPoolForRemoting.submit(() -> Listeners.notify(ComputerListener.class, false, l -> l.onIdle(this)));
```

Comment:
Or just set a flag checked outside the `synchronized` block and run `notify` synchronously then.

Reply:
ok